hub/api/tests/test_objectview.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import hub
+from hub.schema import Tensor, Image, Text, Sequence, SchemaDict, BBox
+import pytest
+
+
+def test_objectview():
+    schema = SchemaDict(
+        {
+            "a": Tensor((20, 20), dtype=int, max_shape=(20, 20)),
+            "b": Sequence(dtype=BBox(dtype=float)),
+            "c": Sequence(
+                dtype=SchemaDict({"d": Sequence((), dtype=Tensor((5, 5), dtype=float))})
+            ),
+            "e": Sequence(
+                dtype={"f": {"g": Tensor(5, dtype=int), "h": Tensor((), dtype=int)}}
+            ),
+        }
+    )
+    ds = hub.Dataset("./nested_seq", shape=(5,), mode="w", schema=schema)
+
+    # dataset view to objectview
+    dv = ds[3:5]
+    dv["c", 0] = {"d": 5 * np.ones((2, 2, 5, 5))}
+    assert (dv[0, "c", 0, "d", 0].compute() == 5 * np.ones((5, 5))).all()
+
+    # dataset view unsqueezed
+    with pytest.raises(IndexError):
+        dv["c", "d"].compute()
+    with pytest.raises(IndexError):
+        dv["c", "d"] = np.ones((2, 3, 3, 5, 5))
+
+    # dataset unsqueezed
+    with pytest.raises(IndexError):
+        ds["c", "d"].compute()
+    with pytest.raises(IndexError):
+        ds["c", "d"] = np.ones((5, 3, 3, 5, 5))
+
+    # tensorview to object view
+    # sequence of tensor
+    ds["b", 0] = 0.5 * np.ones((5, 4))
+    tv = ds["b", 0]
+    tv[0] = 0.3 * np.ones((4,))
+    assert (tv[0].compute() == 0.3 * np.ones((4,))).all()
+
+    # ds to object view
+    assert (ds[3, "c", "d"].compute() == 5 * np.ones((2, 2, 5, 5))).all()
+
+    # Sequence of schemadicts
+    ds[0, "e"] = {"f": {"g": np.ones((3, 5)), "h": np.ones(3)}}
+    ds[0, "e", 0, "f", "h"] = 42
+    # This doesn't work. Add support for this
+    # ds[0, "e", 1]["f", "h"] = 25
+    ds[0, "e"][1]["f"]["h"] = 25
```

Comment:
This might be confusing for the end-user. Are there any particular challenges that prevent this from working?

Reply:
Pushed a fix to allow this indexing